src/app/ipv/middleware.js

```diff
@@ -428,6 +463,38 @@ module.exports = {
       next(error);
     }
   },
+  formHandleUpdateDetailsCheckBox: async (req, res, next) => {
+    try {
+      const { context } = req?.session || "";
+      const pageId = req.session.currentPage;
+
+      const expectedPageId = req.params?.pageId;
+
+      if (expectedPageId && expectedPageId !== pageId) {
+        return await handleUnexpectedPage(req, res, expectedPageId);
+      }
+
+      req.body.journey = getCoiUpdateDetailsJourney(req.body.detailsToUpdate);
+      if (!req.body.journey) {
+        const renderOptions = {
+          pageId,
+          csrfToken: req.csrfToken(),
+          pageErrorState: true,
+          context,
+        };
+
+        if (pageRequiresUserDetails(pageId)) {
+          renderOptions.userDetails = await fetchUserDetails(req);
+        }
+        req.renderOptions = renderOptions;
+        res.render(getIpvPageTemplatePath(sanitize(pageId)), renderOptions);
+      } else {
+        next();
+      }
+    } catch (error) {
+      next(error);
+    }
+  },
```

Comment:
Can we avoid duplicating the unexpected page logic by using the existing `formRadioButtonChecked` middleware function, and then having a separate middleware method that runs afterwards, which can then safely call `getCoiUpdateDetailsJourney()`?

Reply:
@RogVanGDS hmm yeh I can take a look at refactoring :)